### ISSUE TITLE
day/month/year diff calculation spanning DST

### ISSUE THREAD

**Issue by mcraigWTM** [moment/moment-timezone#121]
Not sure if this is a moment bug or a moment-timezone bug (it seems to relate to https://github.com/moment/moment-timezone/issues/120, but that bug itself indicates the reporter believes the report is in the wrong place.)

I've see the same problem with DST and days, months, years calculations.  Given a time in  such as 2014-03-09 10:59:00 America/New_York, diff() says that this is 0 days different than 2014-03-08 10:00:00 America/New_York

This code:
https://gist.github.com/mcraigWTM/32a6569addc2c4327401
gives these results:
$ node gist.js
diffs for day of DST 0
diffs for day of DST 2 1
diffs for months of DST 0
diffs for months of DST 2 1
diffs for year of DST 0
diffs for year of DST 2 1

whereas based on the time of day of each of these it is expected the difference between 10am and 10:59am the next day/month/year would be one day/month/year, the same way the difference between 10am and 11am is one day/month/year.

**Comment by vissermc** [comment]
This is a critical bug that seems to apply only to moment-timezone. 
It does not take into account DST when diffing between days/months/years.
A workaround is to round yourself using Math.round instead of the internal Math.floor that is used.

**Comment by mj1856** [comment]
Ran through @beefsack 's example on jsFiddle, which has moment 2.8.3 and moment-timezone 0.2.2 and got the bad results:
```
144 6
143 5
```

Updated it to the current versions, moment 2.10.2, moment-timezone 0.3.0 and now I get the correct results:
```
144 6
144 6
```

I also ran @mcraigWTM 's sample in node.js with the current versions, and get the output
```
diffs for day of DST  1
diffs for day of DST 2  1
diffs for months of DST  1
diffs for months of DST 2  1
diffs for year of DST  1
diffs for year of DST 2  1
```

Not sure precisely which change fixed this, but it appears to be fixed.
@vissermc  - please let me know if you can still duplicated the problem.

**Comment by andrenascentes** [comment]
@mj1856 I have a similar issue with moment.js version 2.14.1 and momentTimeZone 0.5.5.
I made this jsFiddle so you can it happening https://jsfiddle.net/5ht9rn2g/5/

ps.: Don't know if I should open a new issue or reopen this. If I'm wrong, please let me know and I'll do it right next time :)

**Comment by mj1856** [comment]
@andrenascentes - We've talked about this in the issue I linked above.  I added some commentary there.

Also, the downside with commenting on closed issues is that unless I specifically check my GitHub notifications (of which I sometimes have hundreds), your comment may go unseen.  So yeah, in the future,  prefer opening a new issue.  Worst case, we close it as duplicate, but at least then you know.  😄
